### PR TITLE
Add str property to PurePath in PatLib package 

### PR DESCRIPTION
**Pull request by Arkanayan**
Get string representation of Path with path.str instead of str(Path).

#32807

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/pathlib.py [hunk 1]
```diff
@@ -749,7 +749,10 @@ def __ge__(self, other):
 
     root = property(attrgetter('_root'),
                     doc="""The root of the path, if any.""")
-
+    
+    str = property(methodcaller('__str__',
```
Comment (WildCard65) [inline Lib/pathlib.py:753]: methodcaller is redundant here, remove it (and it's import) and just pass __str__ directly as it's function signature matches the one required by property.

**Review by WildCard65 — CHANGES_REQUESTED** [review]

**Comment by WildCard65** [comment]
To reference a BPO in a Github PR, please modify the PR title to add the following prefix: "bpo-32807" @bedevere-bot will only accept the prefix as lowercase only.

**Comment by JulienPalard** [comment]
Closing as rejected in b.p.o (`.str` is not a Pythonic way to get a string representation of someting). Thanks @Arkanayan for your time and intentions to enhance Python, appreciated,

**Pull request closed without merging**